src/reducers/index.js

```diff
@@ -0,0 +1,42 @@
+const initialState = {
+  restaurants: [],
+  restaurant: {
+    name: '',
+    category: '',
+    address: '',
+  },
+};
+
+const handleReducer = {
+  updateRestaurant: (state, action) => ({
+    ...state,
+    restaurant: {
+      ...state.restaurant,
+      [action.payload.target]: action.payload.value,
+    },
+  }),
+  addRestaurant: (state) => {
+    const { restaurants, restaurant } = state;
+
+    return {
+      ...state,
+      restaurants: [
+        ...restaurants,
+        restaurant,
+      ],
+      restaurant: {
+        name: '',
+        category: '',
+        address: '',
+      },
+    };
+  },
+};
+
+export default function reducer(state = initialState, action) {
+  if (handleReducer[action.type]) {
+    return handleReducer[action.type](state, action);
+  }
+
+  return state;
+}
```

Comment:
```suggestion
export default function reducer(state = initialState, action) {
  if (!handleReducer[action.type]) {
    return state;    
  }

  return handleReducer[action.type](state, action);
}
```

Guard clause를 사용할 때는 진짜 의도를 아래 위치하고, 이 의도를 할 수 없는 경우 빠르게 종료하여 의도를 드러냅니다. 따라서 이렇게 하는게 더 좋을 것 같아요